WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreationThemeLoaderFragment.java

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.ui.accounts.signup;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+
+import org.greenrobot.eventbus.EventBus;
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.action.ThemeAction;
+import org.wordpress.android.fluxc.generated.ThemeActionBuilder;
+import org.wordpress.android.fluxc.store.ThemeStore;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AutoForeground;
+
+import javax.inject.Inject;
+
+public class SiteCreationThemeLoaderFragment extends Fragment {
+    public static final String TAG = "site_creation_theme_loader_fragment_tag";
+
+    public enum ThemesUpdateState {
+        UPDATING,
+        FINISHED,
+        ERROR
+    }
+
+    public static class OnThemeLoadingUpdated implements AutoForeground.ServiceEvent<ThemesUpdateState> {
+        private final ThemesUpdateState state;
+
+        OnThemeLoadingUpdated(ThemesUpdateState state) {
+            this.state = state;
+        }
+
+        @Override
+        public ThemesUpdateState getState() {
+            return state;
+        }
+    }
+
+    @Inject Dispatcher mDispatcher;
+    @Inject ThemeStore mThemeStore;
```

Comment:
It looks like `mThemeStore` is never used.  Is it going to be used in subsequent changes?  If not, can it be removed?

Reply:
That was a leftover from an intermediate state of the code so yes, can be removed, good catch. Done with 20f5064.

Reply:
:man_facepalming: fell victim of this twice not. The ThemeStore injection is actually needed, even without it explicitly referenced elsewhere in the file, otherwise the `onThemesChanged` event listener does not trigger. Will reinstate the injection, sorry for the confusion!

Reply:
Re-instated with 821c4cf.

Reply:
lol